internal/sci/aws/server.go

```diff
@@ -0,0 +1,285 @@
+// Package aws provides an AWS implementation of the Substratus Cloud Interface (SCI)
+package aws
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"os"
+	"time"
+
+	awsSdk "github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/ec2metadata"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/iam"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/sts"
+	"github.com/substratusai/substratus/internal/sci"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+type Server struct {
+	sci.UnimplementedControllerServer
+	OIDCProviderURL string
+	OIDCProviderARN string
+	Clients
+}
+
+type Clients struct {
+	S3Client  *s3.S3
+	IamClient *iam.IAM
+}
+
+func NewAWSServer() (*Server, error) {
+	sess, err := session.NewSession()
+	if err != nil {
+		return nil, fmt.Errorf("failed to create AWS session: %w", err)
+	}
+
+	clusterID, err := getClusterID()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get cluster ID: %w", err)
+	}
+
+	ec2Svc := ec2metadata.New(sess)
+	region, err := getRegion(ec2Svc)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get region: %w", err)
+	}
+
+	stsSvc := sts.New(sess)
+	accountId, err := getAccountID(stsSvc)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get account ID: %w", err)
+	}
+
+	OIDCProviderURL := fmt.Sprintf("oidc.eks.%s.amazonaws.com/id/%s", region, clusterID)
+	OIDCProviderARN := fmt.Sprintf("arn:aws:iam::%s:oidc-provider/%s", accountId, OIDCProviderURL)
+
+	c := &Clients{
+		S3Client:  s3.New(sess),
+		IamClient: iam.New(sess),
+	}
+
+	return &Server{
+		Clients:         *c,
+		OIDCProviderURL: OIDCProviderURL,
+		OIDCProviderARN: OIDCProviderARN,
+	}, nil
+}
+
+func (s *Server) GetObjectMd5(ctx context.Context, req *sci.GetObjectMd5Request) (*sci.GetObjectMd5Response, error) {
+	headResult, err := s.getObjectMetadata(s.Clients.S3Client, req.BucketName, req.ObjectName)
```

Comment:
Prefer to avoid this helper function and move its contents to here. Currently the helper function maps different return conditions into different combo's of nil return values. It would be clearer to do all of that here. There appear to be no other callers of `getObjectMetadata` so it has no re-use benefit.

Reply:
Here's the other caller: https://github.com/substratusai/substratus/pull/186/files#r1292987508

Does that change anything?

Reply:
With the removal of the object existence check, the helper went away. 🤘